### ISSUE TITLE
Support building with `base-4.19.*` (GHC 9.8)

### ISSUE THREAD

**Issue by RyanGlScott** [kquick/kvitable#2]
I tried bumping the upper version bounds on `base` to allow this:

```diff
diff --git a/kvitable.cabal b/kvitable.cabal
index 9c551ad..565c5f8 100644
--- a/kvitable.cabal
+++ b/kvitable.cabal
@@ -53,7 +53,7 @@ library
                      , Data.KVITable.Render.ASCII
                      , Data.KVITable.Render.HTML
   -- other-modules:
-  build-depends:       base >=4.12 && <4.19
+  build-depends:       base >=4.12 && <4.20
                      , containers
                      , lucid  >= 2.9 && < 2.12
                      , microlens >= 0.4 && < 0.5
```

But this results in a volley of `-Wx-partial` warnings:

```
$ cabal build -w ghc-9.8
Resolving dependencies...
Build profile: -w ghc-9.8.1 -O1
In order, the following will be built (use -v for more details):
 - kvitable-1.0.2.1 (lib) (first run)
Configuring library for kvitable-1.0.2.1..
Preprocessing library for kvitable-1.0.2.1..
Building library for kvitable-1.0.2.1..
[1 of 4] Compiling Data.KVITable
[2 of 4] Compiling Data.KVITable.Render
[3 of 4] Compiling Data.KVITable.Render.ASCII

src/Data/KVITable/Render/ASCII.hs:84:26: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
   |
84 |            l = sig sep $ head vals
   |                          ^^^^

src/Data/KVITable/Render/ASCII.hs:126:59: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
126 |     lastFmt = if null hrows then fmtLine [] else hdrFmt $ head $ reverse hrows
    |                                                           ^^^^

src/Data/KVITable/Render/ASCII.hs:170:48: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
170 |       szexts = let subVW = fmtWidth . hdrFmt . head
    |                                                ^^^^

src/Data/KVITable/Render/ASCII.hs:173:71: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
173 |                                               fmtEmptyCols $ hdrFmt $ head sh
    |                                                                       ^^^^

src/Data/KVITable/Render/ASCII.hs:224:78: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
224 |                           then let subl = [ (True, replicate (length $ snd $ head subs) Separator) ]
    |                                                                              ^^^^
[4 of 4] Compiling Data.KVITable.Render.HTML

src/Data/KVITable/Render/HTML.hs:138:33: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
138 |         (HdrLine fmt vals tr) = head nexthdrs -- safe: there were keys
    |                                 ^^^^

src/Data/KVITable/Render/HTML.hs:141:43: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘tail’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Replace it with drop 1, or use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
141 |     in ( (HdrLine fmt' (val : vals) tr) : tail nexthdrs
    |                                           ^^^^

src/Data/KVITable/Render/HTML.hs:169:52: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
169 |                 else L.replicate (length titles) $ head subTtlHdrs
    |                                                    ^^^^

src/Data/KVITable/Render/HTML.hs:171:33: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
171 |       joinHdrs hl = foldl (<>) (head hl) (tail hl)
    |                                 ^^^^

src/Data/KVITable/Render/HTML.hs:171:43: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘tail’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Replace it with drop 1, or use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
171 |       joinHdrs hl = foldl (<>) (head hl) (tail hl)
    |                                           ^^^^
```

It's not immediately clear to me what the best way to resolve these warnings would be.

**Comment by kquick** [comment]
Thanks for the report.  I'll try to resolve these but it may be a couple of days before I have time to work on this.

**Comment by kquick** [comment]
Fixed and released on Hackage as v1.0.3.0
